internal/util/testutil/logging.go

```diff
@@ -50,10 +50,11 @@ func Logger(tb testing.TB) *slog.Logger {
 
 // LevelLogger returns a slog test logger for the given level (which might be dynamic).
 func LevelLogger(tb testing.TB, level slog.Leveler) *slog.Logger {
-	h := logging.NewHandler(NewLogWriter(tb), &logging.NewHandlerOpts{
-		Base:       "console",
-		Level:      level,
-		RemoveTime: true,
+	h := logging.NewHandler(&logWriter{tb: tb}, &logging.NewHandlerOpts{
+		Base:          "console",
+		Level:         level,
+		RemoveTime:    true,
+		CheckMessages: true,
```

Comment:
… but can set it there?.. Why?

Reply:
In the startup code we use the options to set the default logger for the whole integration package, which is then used by `t.Log()`. The logger from testutil however is not used as a default logger, but inside test cases.

Reply:
> In the startup code we use the options to set the default logger for the whole integration package

Not for the integration package. It sets the default logger for `log` and `log/slog` packages. So `log.Print` and `slog.Info` would use it.

> which is then used by `t.Log()`

I'm pretty sure `t.Log()` does not use `log` or `log/slog`.

Try to enable checking in startup.go and check the stacktrace.

Reply:
So it seem to work :)